fnmatch/fnmatch.py

```diff
@@ -64,6 +79,29 @@ def fnmatchcase(name, pat):
     match = _compile_pattern(pat)
     return match(name) is not None
 
+_re_special_chars_map = {i: '\\' + chr(i) for i in b'()[]{}?*+-|^$\\.&~# \t\n\r\v\f'}
+def _re_translate(p, chmap):
+    oldp = p
+    for idx,c in enumerate(p):
+     print('[%d]=%s' % (idx, c))
+     if ord(c) in chmap:
+      newp = p.replace(c, chmap[ord(c)])
+      #print( '%s -> %s' % (oldp, newp) )
+      return newp
+    #print( '%s -> %s' % (oldp, p) )
+    return p
+
+def _re_escape(pattern):
+    """
+    Escape special characters in a string.
+    """
+    print( _re_special_chars_map )
+    if isinstance(pattern, str):
+        #return pattern.translate(_re_special_chars_map)
+        return _re_translate(pattern, _re_special_chars_map )
+    else:
+        pattern = str(pattern, 'latin1')
```

Comment:
Hardcoding an encoding might not be ok here? And why latin1?

Reply:
A copy from cpython implementation. Felt is was better to leave as-is; rather than risk a bug later if/when encoding is supported.

Reply:
Is it ok to just copy from CPython without acnkowledging that? What does its license say?